library/waku.nim

```diff
@@ -0,0 +1,399 @@
+    )
+  except KeyError:
+    jsonResp = errResp(fmt"Problem building the WakuMessage. {getCurrentExceptionMsg()}")
+    return false
+
+  let targetPubSubTopic = if $pubSubTopic == "":
+                            DefaultPubsubTopic
+                          else:
+                            $pubSubTopic
+
+  if node.wakuRelay.isNil():
+    jsonResp = errResp("Can't publish. WakuRelay is not enabled.")
+    return false
+
+  let pubMsgFut = node.wakuRelay.publish(targetPubSubTopic, wakuMessage)
+
+  # With the next loop we convert an asynchronous call into a synchronous one
+  for i in 0 .. timeoutMs:
+    if pubMsgFut.finished():
+      break
+    sleep(1)
+
+  if pubMsgFut.finished():
+    let numPeers = pubMsgFut.read()
+    if numPeers == 0:
+      jsonResp = errResp("Message not sent because no peers found")
+    elif numPeers > 0:
+      # TODO: pending to return a valid message Id (response when all is correct)
+      jsonResp = okResp("hard-coded-message-id")
+
+  else:
+    jsonResp = errResp("Timeout expired")
+
+  return true
+
+proc waku_start() {.dynlib, exportc.} =
+  waitFor node.start()
+
+proc waku_stop() {.dynlib, exportc.} =
+  waitFor node.stop()
+
+proc waku_relay_subscribe(
+                pubSubTopic: cstring,
+                jsonResp: var string): bool
+                {.dynlib, exportc.} =
+  # @params
+  #  topic: Pubsub topic to subscribe to. If empty, it subscribes to the default pubsub topic.
+  if eventCallback == nil:
+    jsonResp = errResp("""Cannot subscribe without a callback.
+Kindly set it with the 'waku_set_event_callback' function""")
+    return false
+
+  if node.wakuRelay.isNil():
+    jsonResp = errResp("Cannot subscribe without Waku Relay enabled.")
+    return false
+
+  node.wakuRelay.subscribe(PubsubTopic($pubSubTopic), PubsubRawHandler(relayEventCallback))
+
+  jsonResp = okResp("true")
+  return true
+
+proc waku_relay_unsubscribe(
+                pubSubTopic: cstring,
+                jsonResp: var string): bool
+                {.dynlib, exportc.} =
+  # @params
+  #  topic: Pubsub topic to subscribe to. If empty, it unsubscribes to the default pubsub topic.
+  if node.wakuRelay == nil:
+    jsonResp = errResp("""Cannot unsubscribe without a callback.
+Kindly set it with the 'waku_set_event_callback' function""")
+    return false
+
+  if node.wakuRelay.isNil():
+    jsonResp = errResp("Cannot unsubscribe without Waku Relay enabled.")
+    return false
+
+  node.wakuRelay.unsubscribeAll(PubsubTopic($pubSubTopic))
+
+  jsonResp = okResp("true")
+  return true
+
+proc waku_connect(peerMultiAddr: cstring,
+                  timeoutMs: uint = 10000,
+                  jsonResp: var string): bool
+                  {.dynlib, exportc.} =
+  # peerMultiAddr: comma-separated list of fully-qualified multiaddresses.
+  let peers = ($peerMultiAddr).split(",").mapIt(strip(it))
+
+  let connectFut = node.connectToNodes(peers, source="static")
+  while not connectFut.finished():
+    poll()
```

Comment:
Wait, where does `timeoutMs` come in here?

Reply:
Indeed, this is something we need to fix from lower levels. I'll add a `TODO` statement here.